python/cudf/cudf/core/dtypes.py

```diff
@@ -103,3 +105,59 @@ def deserialize(cls, header, frames):
             categories_header, categories_frames
         )
         return cls(categories=categories, ordered=ordered)
+
+
+class ListDtype(ExtensionDtype):
+    def __init__(self, value_type):
+        if isinstance(value_type, ListDtype):
+            self._typ = pa.list_(value_type._typ)
+        else:
+            value_type = cudf.utils.dtypes.np_to_pa_dtype(np.dtype(value_type))
+            self._typ = pa.list_(value_type)
+
+    @property
+    def value_type(self):
+        if isinstance(self._typ.value_type, pa.ListType):
+            return ListDtype.from_arrow(self._typ.value_type)
+        else:
+            return np.dtype(self._typ.value_type.to_pandas_dtype()).name
+
+    @property
+    def leaf_type(self):
+        if isinstance(self.value_type, ListDtype):
+            return self.value_type.leaf_type
+        else:
+            return self.value_type
+
+    @property
+    def type(self):
+        return pa.array
```

Comment:
Can we add a `#TODO` here to update this to something more strongly typed in the future? We'll likely need something like `ListDtypeType` similar to `pandas.core.dtypes.dtypes.CategoricalDtypeType`

Reply:
Done